File: app/src/main/java/com/moviepass/model/Provider.java
package com.moviepass.model;

import org.parceler.Parcel;

import java.util.HashMap;

/**
 * Created by anubis on 6/10/17.
 */

@Parcel
public class Provider {

    public String providerName;
    public int theater;
    public String ticketType;
    public HashMap<String, PerformanceInfo> performanceInfo;

    public Provider() {
    }

    public PerformanceInfo getPerformanceInfo(String key) { return performanceInfo.get(key); }

    public int getTheater() { return theater; }

    public boolean ticketTypeIsStandard() {
        return ticketType.matches("STANDARD");
    }

    public boolean ticketTypeIsETicket() {
        return ticketType.matches("E_TICKET");
    }

    public boolean ticketTypeIsSelectSeating() {
        return ticketType.matches("SELECT_SEATING");
    }

    public String getProviderName() {
        return providerName;
    }
}

File: app/src/main/java/com/mobile/model/Provider.kt
package com.mobile.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Provider(var providerName: String? = null,
                    var theater: Int = 0,
                    var ticketType: String? = null,
                    var performanceInfo: Map<String, PerformanceInfo?>? = emptyMap()) : Parcelable
